### PR TITLE
feat: ability to add custom viewrecipe in table expandable row

### PR DESCRIPTION
**Pull request by awesthouse**
## What does this pull request change?
- adds expandableRecipeViewConfig to table blueprint
- adds example expandableRecipeViewConfig to case recipe in signal app
- separate openAsTab and openAsExpandable in tableplugin so you can use both at the same time

## Why is this pull request needed?
Want to be able to add custom view in expandable that may differ from view in open/openAsTab

## Issues related to this change
#258

### CODE REVIEW

File: packages/dm-core-plugins/src/table/TableRow/TableRow.tsx [hunk 1]
```diff
@@ -128,6 +114,17 @@ export function TableRow(props: TTableRow) {
...
         })}
+        {config.functionality?.openAsTab && (
+          <Table.Cell style={{ textAlign: 'center' }}>
+            <Button
+              title="Open"
+              variant="ghost"
+              onClick={() => openItemAsTab(item, index)}
+            >
+              <Icon data={open_in_browser} aria-hidden /> Open
+            </Button>
```
Comment (ingeridhellen) [inline packages/dm-core-plugins/src/table/TableRow/TableRow.tsx:125]: There is several differences here relative to the Table.Cell for openAsExpanable (ie cell style, button variant, button disabled, title, aria-hidden, tooltip...). Is there a specific reason for all those differences?

File: packages/dm-core-plugins/src/table/TablePlugin.tsx [hunk 2]
```diff
@@ -115,13 +115,15 @@ export const TablePlugin = (props: IUIPlugin) => {
       <Table style={{ width: '100%' }}>
         <Table.Head>
           <Table.Row>
-            {(config.functionality?.openAsTab ||
-              config.functionality?.openAsExpandable) && (
+            {config.functionality?.openAsExpandable && (
               <Table.Cell width="80"></Table.Cell>
```
Comment (ingeridhellen) [inline packages/dm-core-plugins/src/table/TablePlugin.tsx:119]: Shouldn't there be an aria-label here as well?

File: packages/dm-core-plugins/src/table/TableRow/TableRow.tsx [hunk 3]
```diff
@@ -178,7 +175,11 @@ export function TableRow(props: TTableRow) {
           <Table.Cell colSpan={getColumnsLength()}>
             <ViewCreator
               idReference={`${idReference}[${item.index}]`}
-              viewConfig={config.views?.[item.index] ?? config.defaultView}
+              viewConfig={
+                config.expandableRecipeViewConfig
+                  ? config.expandableRecipeViewConfig
+                  : config.views?.[item.index] ?? config.defaultView
```
Comment (ingeridhellen) [inline packages/dm-core-plugins/src/table/TableRow/TableRow.tsx:181]: I wonder if the config.views and config.defaultView makes sense anymore. I see they are defined in TTablePluginConfig, but not in TablePluginConfig.json. They don't seem to be used anywhere in our example either. What do you think @eoaksnes?

File: packages/dm-core-plugins/src/table/types.ts [hunk 4]
```diff
@@ -14,6 +18,7 @@ export type TTablePluginConfig = {
   columns: string[]
   showDelete: boolean
```
Comment (ingeridhellen) [inline packages/dm-core-plugins/src/table/types.ts:19]: EditMode and showDelete should be removed from the TTablePluginConfig, as they are no longer part of the blueprint

**Review by ingeridhellen — APPROVED** [review]

**Review by ingeridhellen — CHANGES_REQUESTED** [review]
Meant to just comment, but pressed the approval button by mistake. My bad

File: packages/dm-core-plugins/src/table/TableRow/TableRow.tsx (hunk 3 above)
Comment (ingeridhellen) [reply packages/dm-core-plugins/src/table/TableRow/TableRow.tsx:181]:
```suggestion
                  : { type: 'ViewConfig', scope: 'self' }
```

File: packages/dm-core-plugins/blueprints/table/TablePluginConfig.json [hunk 5]
```diff
@@ -20,6 +20,11 @@
       "default": [],
       "dimensions": "*"
     },
+    {
+      "attributeType": "CORE:ViewConfig",
+      "type": "CORE:BlueprintAttribute",
+      "name": "expandableRecipeViewConfig"
+    },
```
Comment (ingeridhellen) [inline packages/dm-core-plugins/blueprints/table/TablePluginConfig.json:27]: Note to self: Should also add one for openAsTab at some point

File: packages/dm-core-plugins/src/table/TableRow/TableRow.tsx (hunk 1 above)
Comment (awesthouse) [reply packages/dm-core-plugins/src/table/TableRow/TableRow.tsx:125]: disabled is definitely missing and I'll take a look, see what makes sense for each button and do some improvements

File: packages/dm-core-plugins/blueprints/table/TablePluginConfig.json (hunk 5 above)
Comment (awesthouse) [reply packages/dm-core-plugins/blueprints/table/TablePluginConfig.json:27]: But isn't openAsTab just opening the object and should use the default view? I asked Babak and he said yes when I asked if it should just be the ui recipe for the object

File: packages/dm-core-plugins/src/table/types.ts [hunk 6]
```diff
@@ -14,6 +18,7 @@ export type TTablePluginConfig = {
...
   editableColumns?: string[]
+  expandableRecipeViewConfig?: TInlineRecipeViewConfig
```
[the remark this answers is not in the document]
Comment (awesthouse) [reply packages/dm-core-plugins/src/table/types.ts:21]: See that I've used CORE:ViewConfig in blueprint and defined type as CORE:InlineRecipeViewConfig in the ui recipe for case so there's definitely and error, but should it not be InlineRecipeViewConfig?

[the remark this answers is not in the document]
Comment (ingeridhellen) [reply packages/dm-core-plugins/src/table/types.ts:21]: If a blueprint says that an attribute is of type MyBlueprint, and you have another blueprint MyBlueprintExtended that extends MyBlueprint, than an entity with type MyBlueprint can set that attribute to be either of type MyBlueprint or MyBlueprintExtended. So the blueprint and ui recipe is correct.
Maybe the best thing to do here would be
```suggestion
  expandableRecipeViewConfig?: TViewConfig | TInlineRecipeViewConfig | TReferenceViewConfig
```
I see that that is the viewConfig type expected by ViewCreator

File: packages/dm-core-plugins/blueprints/table/TablePluginConfig.json (hunk 5 above)
Comment (ingeridhellen) [reply packages/dm-core-plugins/blueprints/table/TablePluginConfig.json:27]: Mh, okay. Eirik Ola thought this should be configurable. But we can discuss this on the Monday meeting. Either way, it does not need to be fixed right away

File: packages/dm-core-plugins/src/table/TableRow/TableRow.tsx [hunk 7]
```diff
@@ -75,8 +75,11 @@ export function TableRow(props: TTableRow) {
       <Table.Row key={item.key}>
         {config.functionality.openAsExpandable && (
           <Table.Cell>
-            <Tooltip title={item.expanded ? 'Minimize' : 'Expand'}>
+            <Tooltip title={item.expanded ? 'Minimize' : 'Collapse'}>
```
Comment (ingeridhellen) [inline packages/dm-core-plugins/src/table/TableRow/TableRow.tsx:78]: This seems wrong. It should be Expand, right?

File: packages/dm-core-plugins/src/table/TableRow/TableRow.tsx [hunk 8]
```diff
@@ -117,7 +120,7 @@ export function TableRow(props: TTableRow) {
         {config.functionality?.openAsTab && (
           <Table.Cell style={{ textAlign: 'center' }}>
             <Button
-              title="Open"
+              disabled={!item.isSaved}
```
Comment (ingeridhellen) [inline packages/dm-core-plugins/src/table/TableRow/TableRow.tsx:123]: UU is not my area of expertize, but shouldn't there be an aria-label here?

**Review by ingeridhellen — APPROVED** [review]

File: packages/dm-core-plugins/src/table/TableRow/TableRow.tsx (hunk 7 above)
Comment (awesthouse) [reply packages/dm-core-plugins/src/table/TableRow/TableRow.tsx:78]: Oh, I changed the wrong one 🙈 Maybe it is time for summer holidays

File: packages/dm-core-plugins/src/table/TableRow/TableRow.tsx (hunk 8 above)
Comment (awesthouse) [reply packages/dm-core-plugins/src/table/TableRow/TableRow.tsx:123]: It has a label/title in the button, so there's no need 👍🏻 For buttons that only have icons and no text for sr to read is where labeling it is more important

**Pull request merged**